File: app/src/main/java/abandonedstudio/app/cookingapp/Database/Ingredient.java
package abandonedstudio.app.cookingapp.Database;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(tableName = "ingredients", foreignKeys = @ForeignKey(entity = Dish.class, parentColumns = "dishId", childColumns = "dish_id", onDelete = ForeignKey.CASCADE, onUpdate = ForeignKey.CASCADE))
public class Ingredient {

    public Ingredient(String ingredient, int dishId) {
        this.ingredient = ingredient;
        this.dishId = dishId;
    }

    @PrimaryKey(autoGenerate = true)
    private int ingredientId;

    @ColumnInfo(name = "ingredient")
    private String ingredient;

    @ColumnInfo(name = "dish_id")
    private int dishId;

    public int getIngredientId() {
        return ingredientId;
    }

    public void setIngredientId(int ingredientId) {
        this.ingredientId = ingredientId;
    }

    public String getIngredient() {
        return ingredient;
    }

    public int getDishId() {
        return dishId;
    }
}

File: app/src/main/java/abandonedstudio/app/cookingapp/core/data/local/room/ingredient/Ingredient.kt
package abandonedstudio.app.cookingapp.core.data.local.room.ingredient

import abandonedstudio.app.cookingapp.core.data.local.room.dish.Dish
import androidx.room.*

@Entity(
    tableName = "ingredients",
    indices = [Index("dish_id")],
    foreignKeys = [ForeignKey(
        entity = Dish::class,
        parentColumns = ["dishId"],
        childColumns = ["dish_id"],
        onDelete = ForeignKey.CASCADE,
        onUpdate = ForeignKey.CASCADE
    )]
)
data class Ingredient(

    @ColumnInfo(name = "ingredient")
    var ingredient: String,

    @ColumnInfo(name = "dish_id")
    var dishId: Int,

    @PrimaryKey(autoGenerate = true)
    var ingredientId: Int? = null
)
